### ISSUE TITLE
Document does not show function return type which use keyword impl

### ISSUE THREAD

**Issue by ZhouHansen** [rust-lang/cargo#6173]
My function is:
```rust
fn batcher() -> impl Fn(usize) -> () {
// ...
}
```
I run `cargo doc --open`, the document of this function lacks of return type:
```rust
pub fn batcher() -> 
```

cargo version: cargo 1.29.0-nightly

**Comment by alexcrichton** [comment]
Thanks for the report! I think that this may be a bug with `rustdoc` rather than Cargo, so mind opening an issue against https://github.com/rust-lang/rust where the source of `rustdoc` lives? Thanks!

**Comment by ehuss** [comment]
This was fixed in rust/lang-rust#53541 and is available on beta (1.30).

**Comment by alexcrichton** [comment]
Ah ok great! In that case I'll go ahead and close.
